script/test-robots

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+"""
+A script to run a test competition match on every robot in an submissions directory.
+
+A "broken" robot is any robot which:
+- Fails to reach the wait_start function within the 5 second timeout
+  set in wait_until_robots_ready, caused by:
+  a. The robot code throwing an uncaught exception before instantiating the Robot
+  b. The robot code taking over 5 seconds before instantiating the Robot
+- Fails to complete a 1 second match within the subprocess timout
+- Causes the simulator to exit with a non-zero exit code
+"""
+import re
+import sys
+import time
+import shutil
+import argparse
+import tempfile
+import subprocess
+from typing import List, Optional
+from pathlib import Path
+
+# Seconds beyond a normal match duration for the match to be considered stuck
+EXCESS_SUBPROCESS_TIMEOUT = 10
+
+REPO_ROOT = Path(__file__).resolve().parent.parent
+
+sys.path.insert(1, str(REPO_ROOT / 'modules'))
+
+from controller_utils import NUM_ZONES  # isort:skip
+
+
+def run_robot_match(
+    comp_match_script: Path,
+    archives_dir: Path,
+    tla: str,
+    subprocess_timeout: Optional[float],
+) -> bool:
+    try:
+        print(f"Running match for {tla}")  # noqa:T001
+
+        num_empty_zones = NUM_ZONES - 1
+        match_args = [
+            sys.executable,
+            str(comp_match_script),
+            str(archives_dir),
+            '--duration', '1',
+            '--no-record',
+            '--',
+            '0',  # no logs will be retained so a fixed match number is used
+            tla,
+        ] + (['-'] * num_empty_zones)
+
+        subprocess.run(
```

Comment:
You might like https://docs.python.org/3/library/subprocess.html#subprocess.check_call, which deals with `check=True` for you, and has a slightly nicer return which may be useful.

Reply:
> Note: Do not use stdout=PIPE or stderr=PIPE with this function. The child process will block if it generates enough output to a pipe to fill up the OS pipe buffer as the pipes are not being read from. 

Yeah best leave as-is